src/core/dynamic_batch_scheduler.cc

```diff
@@ -152,18 +158,28 @@ DynamicBatchScheduler::~DynamicBatchScheduler()
   }
 }
 
-Status
-DynamicBatchScheduler::Enqueue(std::unique_ptr<InferenceRequest>& request)
-{
-  // Queue timer starts at the beginning of the queueing and
-  // scheduling process
-  request->CaptureQueueStartNs();
-  INFER_TRACE_ACTIVITY(
-      request->Trace(), TRITONSERVER_TRACE_QUEUE_START,
-      request->QueueStartNs());
+Status DynamicBatchScheduler::Enqueue(
+    std::unique_ptr<InferenceRequest>& request) {
+  // If queue start timestamp hasn't been set, queue timer starts at
+  // the beginning of the queueing and scheduling process. Otherwise,
+  // dynamic batcher is used as component of another batcher and should not
+  // overwrite the queue start timestamp.
+  if (request->QueueStartNs() == 0) {
+    request->CaptureQueueStartNs();
+    INFER_TRACE_ACTIVITY(request->Trace(), TRITONSERVER_TRACE_QUEUE_START,
+                         request->QueueStartNs());
+  }
+
+  // Record time at the beginning of the batcher queueing. In the case of
+  // oldest sequence batcher, this will overwrite the value that was previously
+  // set by sequence batcher, which is okay as by this point, the previous
+  // batcher won't be needing this value and it can be safely reused by
+  // the dynamic batcher.
+  request->CaptureBatcherStartNs();
+
 #ifdef TRITON_ENABLE_TRACING
-  request->TraceInputTensors(
-      TRITONSERVER_TRACE_TENSOR_QUEUE_INPUT, "DynamicBatchScheduler Enqueue");
+  request->TraceInputTensors(TRITONSERVER_TRACE_TENSOR_QUEUE_INPUT,
```

Comment:
Move this to just after line 170, it needs to be inside of the conditional above.